### PR TITLE
[Trunk] Rename experiments package to private-apis

### PR DESCRIPTION
**Pull request by adamziel**
## What?

Renames the @wordpress/experiments package to @wordpress/private-apis to avoid confusion between internal APIs and experimental APIs. See https://github.com/WordPress/gutenberg/issues/47785

## Testing Instructions
Confirm the checks pass

cc @youknowriad @mcsf @peterwilsoncc

### CODE REVIEW

**Comment by adamziel** [comment]
@geriux any idea why the mobile test suite fails with this PR?

```
FAIL packages/edit-post/src/components/visual-editor/test/index.native.js
  ● Test suite failed to run

    TypeError: Cannot use 'in' operator to search for 'WebkitAnimation' in undefined

      at getVendorPrefixedEventName (node_modules/react-dom/cjs/react-dom.development.js:8797:58)
      at node_modules/react-dom/cjs/react-dom.development.js:8805:21
      at Object.<anonymous> (node_modules/react-dom/cjs/react-dom.development.js:29867:5)
      at Object.<anonymous> (node_modules/react-dom/index.js:37:20)
```

**Comment by geriux** [comment]
> @geriux any idea why the mobile test suite fails with this PR?

I'll check it out 👍

**Comment by geriux** [comment]
Hey @adamziel 👋  We need to rename `packages/block-editor/src/experiments.native.js` to `packages/block-editor/src/private-apis.native.js` With that change, unit tests and E2E test will work. Let me know if you need anything else.

**Comment by adamziel** [comment]
Thank you @geriux ! It works now

**Review by ntsekouras — APPROVED** [review]
LGTM, thanks Adam!

**Comment by ntsekouras** [comment]
Cherry-picked this PR to the wp/6.2 branch.

File: docs/contributors/code/coding-guidelines.md [hunk 1]
```diff
@@ -238,14 +238,14 @@ export function toggleFeature( scope, featureName ) {
...
+#### Use the `lock()` and `unlock()` API from `@wordpress/private-apis` to privately export almost anything
 
 Each `@wordpress` package wanting to privately access or expose experimental APIs can
-do so by opting-in to `@wordpress/experiments`:
+do so by opting-in to `@wordpress/private-apis`:
 
 ```js
-// In packages/block-editor/experiments.js:
-import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/experiments';
+// In packages/block-editorwordpress/private-apis.js:
```
Comment (peterwilsoncc) [inline docs/contributors/code/coding-guidelines.md:247]: @adamziel Sorry for the post-merge pick up, should this line read `packages/block-editor/private-apis.js`

I noticed it elsewhere in the PR too.

Comment (ntsekouras) [reply docs/contributors/code/coding-guidelines.md:247]: Thanks! https://github.com/WordPress/gutenberg/pull/47953

**Pull request merged**